src/arrayMethodJoin.js

```diff
@@ -3,9 +3,20 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    const newSeparator = separator === undefined ? ',' : separator;
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        result = result + this[i];
+      }
+      result = result + newSeparator;
+    }
+
+    return result.substring(0, result.length - String(newSeparator).length);
```

Comment:
it's better to check in loop if `i === this.length - 1` and don't add separator here